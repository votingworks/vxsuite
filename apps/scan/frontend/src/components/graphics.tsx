import React from 'react';
import styled from 'styled-components';
import { Graphic } from '@votingworks/ui';

export const PlaceholderGraphic = styled.div`
  margin: 0 auto 1rem;
  border-radius: 1rem;
  background: #6638b6;
  width: 250px;
  height: 250px;
`;

export function QuestionCircle(): JSX.Element {
  return (
    <Graphic
      src="/assets/question-circle.svg"
      alt="question-circle"
      aria-hidden
    />
  );
}

export function DoNotEnter(): JSX.Element {
  return (
    <Graphic src="/assets/do-not-enter.svg" alt="do-not-enter" aria-hidden />
  );
}

export function CircleCheck(): JSX.Element {
  return (
    <Graphic src="/assets/check-circle.svg" alt="check-circle" aria-hidden />
  );
}

export function TimesCircle(): JSX.Element {
  return (
    <Graphic
      src="/assets/times-circle.svg"
      alt="times-circle"
      aria-hidden
      style={{ height: '400px' }}
    />
  );
}

export function ExclamationTriangle(): JSX.Element {
  return (
    <Graphic
      src="/assets/exclamation-triangle.svg"
      alt="exclamation-triangle"
      aria-hidden
    />
  );
}
