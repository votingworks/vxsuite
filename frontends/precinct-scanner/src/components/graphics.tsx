import React from 'react';
import styled from 'styled-components';

export const PlaceholderGraphic = styled.div`
  margin: 0 auto 1rem;
  border-radius: 1rem;
  background: #6638b6;
  width: 250px;
  height: 250px;
`;

export const Graphic = styled.img`
  margin: 0 auto 1rem;
  width: 250px;
`;

export function QuestionCircle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/question-circle.svg`}
      alt="question-circle"
      aria-hidden
    />
  );
}

export function DoNotEnter(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/do-not-enter.svg`}
      alt="do-not-enter"
      aria-hidden
    />
  );
}

export function CircleCheck(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/check-circle.svg`}
      alt="check-circle"
      aria-hidden
    />
  );
}

export function TimesCircle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/times-circle.svg`}
      alt="times-circle"
      aria-hidden
      style={{ height: '400px' }}
    />
  );
}

export function ExclamationTriangle(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/exclamation-triangle.svg`}
      alt="exclamation-triangle"
      aria-hidden
    />
  );
}

export function InsertBallot(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/insert-ballot.svg`}
      alt="insert-ballot"
      aria-hidden
      style={{ width: '600px' }}
    />
  );
}

export function IndeterminateProgressBar(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/indeterminate-progress-bar.svg`}
      alt="indeterminate-progress-bar"
      aria-hidden
      style={{ width: '400px' }}
    />
  );
}

export function RotateCard(): JSX.Element {
  return (
    <Graphic
      src={`${process.env.PUBLIC_URL}/assets/rotate-card.svg`}
      alt="rotate-card"
      aria-hidden
      style={{ width: '300px' }}
    />
  );
}
