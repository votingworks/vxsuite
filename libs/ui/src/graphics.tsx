import React from 'react';
import styled from 'styled-components';

export const Graphic = styled.img`
  margin: 0 auto 1rem;
  width: 250px;
`;

const UsbImageStyles = styled.img`
  margin-right: auto;
  margin-left: auto;
  height: 200px;
`;

export function UsbImage(): JSX.Element {
  return <UsbImageStyles src="/assets/usb-drive.svg" alt="Insert USB Image" />;
}

export function IndeterminateProgressBar(): JSX.Element {
  return (
    <Graphic
      src="/assets/indeterminate-progress-bar.svg"
      alt="indeterminate-progress-bar"
      aria-hidden
      style={{ width: '400px' }}
    />
  );
}
