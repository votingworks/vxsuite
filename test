#!/usr/bin/env bun

import { readFile } from 'fs/promises';
import { join } from 'path';

const { $ } = Bun;
export {};

const workspacePackageJson = await readFile(
  join(__dirname, 'package.json'),
  'utf8'
);
const { workspaces } = JSON.parse(workspacePackageJson);
const WIDTH = 100;

const results: { pkgPath: string; status: 'success' | 'failure' }[] = [];
let totalPassed = 0;
let totalFailed = 0;

for (const pkgPath of workspaces) {
  const tests = box(
    (async function* () {
      yield `→ bun run test --cwd ${pkgPath}`;
      const bunTest = $`bun run test --cwd ${pkgPath}`.quiet();

      const bunTestResult = await bunTest;

      yield* bunTestResult.stderr.toString('utf-8').split('\n');
      yield `${bunTestResult.exitCode === 0 ? '✅' : '❌'} ${pkgPath}`;

      if (bunTestResult.exitCode === 0) {
        results.push({ pkgPath, status: 'success' });
      } else {
        results.push({ pkgPath, status: 'failure' });
      }
    })()
  );

  let passed = 0;
  let failed = 0;

  for await (const line of tests) {
    const passedMatch = line.match(/(\d+) pass/);
    const failedMatch = line.match(/(\d+) fail/);

    if (passedMatch) {
      passed = parseInt(passedMatch[1], 10);
    }

    if (failedMatch) {
      failed = parseInt(failedMatch[1], 10);
    }

    console.log(line);
  }

  totalPassed += passed;
  totalFailed += failed;
}

console.log(
  `\n${totalPassed + totalFailed} tests: ${totalPassed} passed, ${totalFailed} failed`
);

for (const { pkgPath, status } of results) {
  console.log(`${status === 'success' ? '✅' : '❌'} ${pkgPath}`);
}

async function* box(lines: AsyncIterable<string>): AsyncIterable<string> {
  yield* lines;
}
