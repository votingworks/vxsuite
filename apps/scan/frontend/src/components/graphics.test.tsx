// This is a temporary test to be removed once SVG graphics are created.
import React from 'react';
import { render } from '../../test/react_testing_library';
import { PlaceholderGraphic } from './graphics';

test('Renders Absolute top right', () => {
  const { container } = render(<PlaceholderGraphic />);
  expect(container).toMatchSnapshot();
});
