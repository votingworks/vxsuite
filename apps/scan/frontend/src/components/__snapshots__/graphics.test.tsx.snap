// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Absolute top right 1`] = `
.c0 {
  margin: 0 auto 1rem;
  border-radius: 1rem;
  background: #6638b6;
  width: 250px;
  height: 250px;
}

<div>
  <div
    class="c0"
  />
</div>
`;
